internal/blockchain/ethereum/ethereum.go

```diff
@@ -718,10 +717,15 @@ func (e *Ethereum) InvokeContract(ctx context.Context, nsOpID string, signingKey
 	if err != nil {
 		return err
 	}
-	abi, orderedInput, err := e.prepareRequest(ctx, method, input)
+	var errors []*fftypes.FFIError
+	if errorDefs := options["errors"]; errorDefs != nil {
+		errors = errorDefs.([]*fftypes.FFIError)
+	}
+	abi, errorsAbi, orderedInput, err := e.prepareRequest(ctx, method, errors, input)
 	if err != nil {
 		return err
 	}
+	options["errors"] = errorsAbi
```

Comment:
in fact that was the route I took originally but then realized that the `errors` as a construct in the interface definition doesn't apply universally to other protocols such as Fabric. So instead of forcing that as part of the plugin API, I decided to use the options so it fits with this being an "optional" feature. But if you have a strong objection to this approach, I can certainly move to making it part of the `InvokeContract()` signature out of the `options` parameter

Reply:
upon further discussing this with @nguyer and @peterbroadhurst, I realized that the `errors` construct is already part of the standard FFI API because of the earlier work in FFTM: https://github.com/hyperledger/firefly-transaction-manager/blob/main/pkg/ffcapi/api.go#L202. so I'll make the update as suggested above by Nicko